### PR TITLE
Move methods that are not used outside of class to private section in app/controllers folder

### PR DESCRIPTION
**Pull request by alkeshrails**
Use private section for methods that are not used outside of class.
link : https://aicrowd.monday.com/boards/363093650/pulses/420516667

### CODE REVIEW

File: app/controllers/api/external_graders_controller.rb [hunk 1]
```diff
@@ -245,105 +218,11 @@ def post_challenge(challenge, params)
...
-    return false if participant_leaderboard.nil?
-    if participant_leaderboard.score >= min_score
-      return true
-    else
-      return false
-    end
-  end
-
+  # Unused method
   def validate_s3_key(s3_key)
```
Comment (piotrekpasciak) [inline app/controllers/api/external_graders_controller.rb]: Please remove that method.

File: app/controllers/challenge_rules_controller.rb [hunk 2]
```diff
@@ -6,6 +6,7 @@ def show; end
 
   private
 
+  # Unused method
   def challenge_rules_params
```
Comment (piotrekpasciak) [inline app/controllers/challenge_rules_controller.rb]: Just remove the method.

File: app/controllers/participation_terms_controller.rb [hunk 3]
```diff
@@ -8,6 +8,7 @@ def show
 
   private
 
+  # Unused method
   def participation_terms_params
```
Comment (piotrekpasciak) [inline app/controllers/participation_terms_controller.rb]: Remove it.

File: app/controllers/api/old_submissions_controller.rb [hunk 4]
```diff
@@ -39,6 +39,7 @@ def index
   private
 
   def set_submissions(challenge_id, grading_status, after, challenge_round_id)
+    # can manage it in multi level if-else loop
```
Comment (piotrekpasciak) [inline app/controllers/api/old_submissions_controller.rb]: You're right this method seems to be major overkill. You can leave it  for now, but if you really want to refactor it, do it in separate PR.

**Comment by piotrekpasciak** [comment]
@alkeshrails 
Could you also take a look on Codeclimate issues? Would it be easy to fix them?
https://codeclimate.com/github/AIcrowd/AIcrowd/pull/1130

File: app/controllers/api/old_submissions_controller.rb (hunk 4 above)
Comment (alkeshrails) [reply app/controllers/api/old_submissions_controller.rb]: Okay I will handle it in a separate PR. creating ticket for this.

**Comment by alkeshrails** [comment]
This PR is created by fork of repo. I create a new PR for same to app repo.
